apps/spruce/src/utils/tasks/estimatedActivatedTasks.ts

```diff
@@ -0,0 +1,139 @@
+import { GeneratedTaskCountResults, VariantTask } from "gql/generated/types";
+import { versionSelectedTasks } from "hooks/useVersionTaskStatusSelect";
+import { VariantTasksState } from "pages/configurePatch/configurePatchCore/useConfigurePatch/types";
+
+type ValidInputTypes = Set<string> | versionSelectedTasks | VariantTasksState;
+
+/**
+ * `getNumEstimatedActivatedTasks` Computes the number of tasks that are estimated to be scheduled as a result of a scheduling / restart operation. This function
+ * requires a separate helper function to compute this number for every type of `inputType` that can get passed in.
+ * @param generatedTaskCounts - An array of objects containing info on the estimated number of tasks that each generator task (a task that calls generate.tasks)
+ * will schedule, including the `taskId`, `buildVariantName`, `taskName`, and `estimatedTasks`.
+ * @param inputType - The input data structure representing the state of the selected tasks in the UI. This can be a Set, versionSelectedTasks, or VariantTasksState
+ * type depending on whether this is being called from the "Schedule Tasks", "Restart Version", or "Configure Patch" components, respectively.
+ * @param existingVariantsTasks - An optional parameter that represents the initial state of selected tasks in the "Configure Patch" page.
+ * @returns The total sum of all selected tasks and the `estimatedTasks` value of each selected task's corresponding generated tasks estimate, if it exists.
+ */
+export const getNumEstimatedActivatedTasks = (
+  generatedTaskCounts: GeneratedTaskCountResults[],
+  inputType: ValidInputTypes,
+  existingVariantsTasks?: Array<VariantTask>,
+): number => {
+  // Compute for "Schedule Tasks"
+  if (inputType instanceof Set) {
+    return sumActivatedTasksInSet(inputType, generatedTaskCounts);
+  }
+  // Compute for "Restart Version"
+  if (existingVariantsTasks && isVariantTasksState(inputType)) {
+    return sumActivatedTasksInVariantsTasks(
+      inputType,
+      generatedTaskCounts,
+      existingVariantsTasks,
+    );
+  }
+  // Compute for "Configure Patch"
+  if (isVersionSelectedTasks(inputType)) {
+    return sumActivatedTasksInSelectedTasks(inputType, generatedTaskCounts);
+  }
+  throw new Error("Unrecognized input type for counting generated tasks");
+};
```

Comment:
Sorry, I take this back the logic for diffing the different types seems error-prone. I think we should use the underlying methods directly where they are needed.